signing/rotate_gpg_signing_key.rst

```diff
@@ -22,6 +23,98 @@ Generate a new signing subkey
+
+.. _Autograph: https://github.com/mozilla-services/autograph
+
+Deployment Part 1: Testing the new signing subkey
+-------------------------------------------------
+
+The new signing subkey will be deployed to the production instance of `Autograph`_, which means it must be tested through a production worker and repository. The best and safest way to do this is against the `Adhoc Signing`_ repository - which will not impact any users or developers if something goes wrong. You can do this as follows:
+
+1) Update the `gpgPubkey` entry for `firefoxci-adhoc-3` in the `cloudops-infra` repository.
+
+    a) Extract and decode the current base64 entry with a command such as::
+
+        echo "<base64 string>" | base64 -d > current.key
+
+    b) Create a new gpg keyring and import the current key::
+
+        mkdir new-keyring
+        gpg --homedir new-keyring --import current.key
+
+    c) Add the new public key to the keyring::
+
+        gpg --homedir new-keyring --import new-public.key
+
+    d) Export *all* the keys to a new armored file::
+
+        gpg --homedir new-keyring --export --armor > new.key
+
+    e) Add the header from the existing KEY file to `new.key`, with the new sub key information. For example, in the 2023 rotations we added the following (the last line is the new subkey information)::
+
+        This file contains the public PGP key that is used to sign builds and
+        artifacts of Mozilla projects (such as Firefox and Thunderbird).
+
+        Please realize that this file itself or the public key servers may be
+        compromised.  You are encouraged to validate the authenticity of these keys in
+        an out-of-band manner.
+
+        Mozilla users: pgp < KEY
+
+        pub   rsa4096 2015-07-17 [SC]
+              14F26682D0916CDD81E37B6D61B7B526D98F0353
+        uid           [  full  ] Mozilla Software Releases <release@mozilla.com>
+        sub   rsa4096 2015-07-17 [S] [expired: 2017-07-16]
+        sub   rsa4096 2017-06-22 [S] [expired: 2019-06-22]
+        sub   rsa4096 2019-05-30 [S] [expires: 2021-05-29]
+        sub   rsa4096 2021-05-17 [S] [expires: 2023-05-17]
+        sub   rsa4096 2023-05-05 [S] [expires: 2025-05-04]
+
+    f) base64 encode `new.key` and update the `gpgPubkey` entry in the `cloudops-infra` repo for `firefoxci-adhoc-3`::
+
+        cat new.key | base64 -w0
+
+    g) Open a Pull Request with the changes; wait for it to get merged.
+
+2) Update the GPG username and password for `firefox-ci-adhoc-3` in the relengworker SOPS repository.
+3) Deploy production scriptworkers to pick up the changes you made above.
+4) Open a PR with a new signing manifest in the `Adhoc Signing`_ repository that will sign with GPG (note: when it does a test signing as part of opening a PR, it will not be using the new keys, because PRs use dep certs).
```

Comment:
Should we link to an example PR here so we can basically re-use the same manifest?
(Alternatively, if we keep the manifest around, we just have to trigger promote, no need for a new PR.)

Reply:
That's a good idea, let's do it!